src/theme-context-provider.tsx

```diff
@@ -171,7 +161,25 @@ const ThemeContextProvider: React.FC<Props> = ({theme, children, as, withoutStyl
         [colors]
     );
 
+    // TODO: create CSS vars for size and lineHeight (https://jira.tid.es/browse/WEB-1929)
+    const textPresetsVars = React.useMemo(() => {
+        // Get an object mapping textPresets tokens to objects containing the token's weight
+        // For example, {title1: {weight: '700'}}
+        const textTokens = Object.keys(contextTheme.textPresets) as Array<keyof TextPresetsConfig>;
+        const tokenValues = textTokens.map((token) => {
+            // Map light/regular/medium/bold to valid css fontWeight values
+            return {[token]: {weight: String(mapToWeight[contextTheme.textPresets[token].weight])}};
+        });
+
+        const textPresetsVars = Object.assign({}, ...tokenValues) as {
+            [key in keyof TextPresetsConfig]: {weight: string};
+        };
```

Comment:
Why is the assign needed?

Reply:
Discussed offline. TokenValues is an array, we need to merge it into an object. That's why we use Object.assign()